### ISSUE TITLE
Add a mode to ROIAlign for implementing tf.crop_and_resize?

### ISSUE THREAD

**Issue by jiafatom** [onnx/onnx#2100]
In tensorflow, many object detection models use tf.crop_and_resize to extract ROI features. One similar op in onnx doing this is ROIAlign. But they are now behaving differently in three aspects:
(1) the order of input image tensor differs, ROIAlign [x1,y1,x2,y2], tf.crop_and_resize [y1,x1,y2,x2]
(2) ROIAlign rois value is integer, tf.crop_and_resize is normalized by roi size
(3) The sampling technique is different. 

(3) is the most important because we cannot use ROIAlign to implement crop_and_resize, then the conversion of those pretrained object model cannot get correct result. Since these two ops are doing similar things, I want to propose **adding a new mode for ROIAlign** which uses the same sampling technique as crop_and_resize. Then ROIAlign has two modes, one is crop_and_resize, the other is the existing algorithm.

Thank you.

**Comment by wschin** [comment]
[crop_and_resize](https://www.tensorflow.org/api_docs/python/tf/image/crop_and_resize) is a combination of two steps: (1) Crop and (2) Resize. Couldn't we use `Crop` and `Resize` from ONNX to compose it?

**Comment by fuzzyBatman** [comment]
@jiafatom Are you adding the support for this op (CropAndResize) to ONNX?

**Comment by jiafatom** [comment]
> @jiafatom Are you adding the support for this op (CropAndResize) to ONNX?

yes

**Comment by jiafatom** [comment]
> [crop_and_resize](https://www.tensorflow.org/api_docs/python/tf/image/crop_and_resize) is a combination of two steps: (1) Crop and (2) Resize. Couldn't we use `Crop` and `Resize` from ONNX to compose it?

synced offline, we agree on adding a mode for `ROIAlign`.

**Comment by fuzzyBatman** [comment]
> > @jiafatom Are you adding the support for this op (CropAndResize) to ONNX?
> 
> yes

Thank you! I was thinking of working on it.
However, you seem to have a better idea and a head start.
Waiting for the op support :)

**Comment by jiafatom** [comment]
With offline discussion, the current solution is to add CropAndResize op rather than adding a mode.
See [PR](https://github.com/onnx/onnx/pull/2142)
